src/scandeval/generation.py

```diff
@@ -165,79 +192,215 @@ def generate_single_iteration(
             The configuration of the dataset.
         benchmark_config:
             The configuration of the benchmark.
+        cache_path:
+            The path to the model output cache.
 
     Returns:
         A list of dictionaries containing the scores for each metric.
     """
-    # Tokens used in generation to know when generation is finished
-    stopping_criteria = get_generation_stopping_criteria(
-        tokenizer=tokenizer, model=model
-    )
+    # Load the model output cache
+    with cache_path.open() as f:
+        json_cache = json.load(f)
+        cache: dict[str, dict[str, CachedModelOutput]] = defaultdict(dict)
+        if not hasattr(sys, "_called_from_test"):
```

Comment:
This seems a bit hacky way of not using the cache while testing. Any reason why you don't use a flag. An alternative would be to input the cache as a argument. E.g. have a ModelOutputCache object with a method like `get_cache()`. Then during testing you can pass a DummyCache (dependency injection).

Reply:
It's mostly to maintain consistency with the rest of the code base, since I need it in a handful of functions. Seems a bit simpler than juggling around different flags everywhere 🙂

Reply:
But I've refactored many of the code blocks into separate functions now, so the cache functionality can still be tested